src/group_theory/submonoid/operations.lean

```diff
@@ -896,3 +896,146 @@ instance [mul_action M' α] [has_faithful_scalar M' α] (S : submonoid M') :
 end submonoid
 
 end actions
+
+-- For the pointwise inverse for submonoids of groups,
+-- please refer to `group_theory.submonoid.pointwise`.
+section pointwise_inverse
+
+namespace submonoid
+
+@[to_additive] noncomputable
+instance {M : Type*} [monoid M] : group (is_unit.submonoid M) :=
+{ inv := λ x, ⟨_, (x.prop.unit⁻¹).is_unit⟩,
+  mul_left_inv := λ x, subtype.eq x.prop.unit.inv_val,
+ ..(show monoid (is_unit.submonoid M), by apply_instance) }
+
+
+@[to_additive] noncomputable
+instance {M : Type*} [comm_monoid M] : comm_group (is_unit.submonoid M) :=
+{ mul_comm := λ a b, mul_comm a b,
+ ..(show group (is_unit.submonoid M), by apply_instance) }
+
+section monoid
+
+variables {M' : Type*} [monoid M'] (S' : submonoid M')
+
+/-- `S'.left_inv` is the submonoid containing all the left inverses of `S'`. -/
```

Comment:
I'd be tempted to put all this stuff in a new file

Reply:
Because that gives a nice place to write a module docstring, and the file will get quite big if someone comes along wanting right inverses too.

Reply:
Moved to `group_theory.submonoid.inverses`.